composer/callbacks/image_visualizer.py

```
@@ -0,0 +1,189 @@
+            element is the target. Default: ``1``.
+    """
+
+    def __init__(self,
+                 interval: str = '100ba',
+                 mode: str = 'input',
+                 num_images: int = 8,
+                 input_key: Union[str, int, Tuple[Callable, Callable], Any] = 0,
+                 target_key: Union[str, int, Tuple[Callable, Callable], Any] = 1):
+        self.interval = interval
+        self.mode = mode
+        self.num_images = num_images
+        self.input_key = input_key
+        self.target_key = target_key
+
+        try:
+            import wandb
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='wandb',
+                                                conda_package='wandb',
+                                                conda_channel='conda-forge') from e
+        del wandb  # unused
+
+        # Check that the output mode is valid
+        if self.mode.lower() not in ['input', 'segmentation']:
+            raise ValueError(f'Invalid mode: {mode}')
+
+        # Check that the interval timestring is parsable and convert into time object
+        try:
+            self.interval = Time.from_timestring(interval)
+        except ValueError as error:
+            raise ValueError(f'Invalid time string for parameter interval') from error
+
+        # Verify that the interval has supported units.
+        if self.interval.unit not in [TimeUnit.BATCH, TimeUnit.EPOCH]:
+            raise ValueError(f'Invalid time unit for parameter interval: '
+                             f'{self.interval.unit}')
+
+    def _log_inputs(self, state: State, logger: Logger, key: str):
+        inputs = state.batch_get_item(key=self.input_key)
+        if not isinstance(inputs, torch.Tensor):
+            raise NotImplementedError('Multiple input tensors not supported yet')
+        # Verify inputs is a valid shape for conversion to an image
+        if _check_for_image_format(inputs):
```

Should we raise an exception if the batch is not in image format?

---

I decided not to, as training presumably will still work fine unless there's other unrelated bugs, just images won't be logged. I don't want to interrupt a training run if someone is just using the callback when they shouldn't. Happy to make the change though if you think we should raise an exception here.